File: app/src/main/java/com/themovielist/moviedetail/trailer/MovieTrailerListDialogPresenter.java
package com.themovielist.moviedetail.trailer;

import com.themovielist.model.MovieTrailerModel;

import java.util.List;

import javax.inject.Inject;

public class MovieTrailerListDialogPresenter implements MovieTrailerListDialogContract.Presenter {
    private MovieTrailerListDialogContract.View mView;

    @Inject
    public MovieTrailerListDialogPresenter() { }

    @Override
    public void setView(MovieTrailerListDialogContract.View view) {
        mView = view;
    }


    @Override
    public void start(List<MovieTrailerModel> movieTrailerList) {
        mView.showTrailersIntoList(movieTrailerList);
    }
}

File: app/src/main/java/com/themovielist/moviedetail/trailer/MovieTrailerListDialogPresenter.kt
package com.themovielist.moviedetail.trailer

import com.themovielist.model.MovieTrailerModel

import javax.inject.Inject

class MovieTrailerListDialogPresenter @Inject
constructor() : MovieTrailerListDialogContract.Presenter {
    private var mView: MovieTrailerListDialogContract.View? = null

    override fun setView(view: MovieTrailerListDialogContract.View) {
        mView = view
    }


    override fun start(movieTrailerList: List<MovieTrailerModel>) {
        mView!!.showTrailersIntoList(movieTrailerList)
    }
}
